src/compiler/checker.ts

```diff
@@ -14559,6 +14558,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return links.resolvedSignature;
     }
 
+    function tryGetConstructorDeclaration(declaration: SignatureDeclaration | JSDocSignature) {
+        const node = isJSDocSignature(declaration) ? getEffectiveJSDocHost(declaration) : declaration;

```

Comment:
isJSDocSignature is checking for declarations like `@callback`, right? Or is it closure signature syntax like `function(string): string`?

Either way, do we have tests that fail without this check? I don't see them in the new test, so if not, we'll need to expand it.

Reply:
I think the key here is to check if the host is a constructor to get the local type parameters of the class.